salt/orchestrate/edx/backup.sls

```diff
@@ -61,18 +63,29 @@ deploy_backup_instance_to_{{ ENVIRONMENT }}:
         - file: load_backup_host_cloud_profile
         - boto_iam_role: ensure_instance_profile_exists_for_backups
 
-format_backup_drive:
+{# Duplicity requires an archive directory otherwise it will have to create it and download files
+from s3 buckets when called. In order to accomodate that, we have an EBS volume that will be mounted
+by the ephemeral instance that is destroyed once backups are complete. #}
+{% set instance_id = salt.boto_ec2.find_instances('name=backups-{}'.format(ENVIRONMENT)) %}
+{% if instance_id %}
+create_attach_backup_volume:
   salt.function:
-    - tgt: 'G@roles:backups and G@environment:{{ ENVIRONMENT }}'
-    - tgt_type: compound
-    - name: state.single
+    - name: saltutil.runner
     - arg:
-        - blockdev.formatted
+        - cloud.action
     - kwarg:
-        name: /dev/xvdb
-        fs_type: ext4
+        func: ec2.create_attach_volume
+        kwargs:
+          instance_id: {{ instance_id }}
```

Comment:
Looking again, we don't want to pass `instance_id`, instead we want to pass the instance _name_ (e.g. backups-mitx-rp) because the instance_id will be looked up based on that name if it's not explicitly provided. Since we are creating the instance as part of this orchestrate state we won'd have any way to resolve the instance_id ahead of time because it doesn't yet exist when the state file is being rendered.